rasa/core/policies/ted_policy.py

```diff
@@ -347,22 +355,32 @@ def __init__(
         self.only_e2e = TEXT in self.fake_features and INTENT not in self.fake_features
 
         self._label_data: Optional[RasaModelData] = None
-        self.data_example: Optional[Dict[Text, List[np.ndarray]]] = None
+        self.data_example: Optional[Dict[Text, Dict[Text, List[FeatureArray]]]] = None
 
         self.tmp_checkpoint_dir = None
         if self.config[CHECKPOINT_MODEL]:
             self.tmp_checkpoint_dir = Path(rasa.utils.io.create_temporary_directory())
 
+    @staticmethod
+    def model_class() -> Type[RasaModel]:
+        return TED
+
+    @classmethod
+    def _metadata_filename(cls) -> Optional[Text]:
+        return SAVE_MODEL_FILE_NAME
+
     def _load_params(self, **kwargs: Dict[Text, Any]) -> None:
         new_config = rasa.utils.train_utils.check_core_deprecated_options(kwargs)
         self.config = rasa.utils.train_utils.override_defaults(
             self.defaults, new_config
         )
 
-        self.config = rasa.utils.train_utils.update_confidence_type(self.config)
+        self._auto_update_configuration()
 
+    def _auto_update_configuration(self) -> None:
+        """Take care of deprecations and compatibility of parameters."""
+        self.config = rasa.utils.train_utils.update_confidence_type(self.config)
```

Comment:
Why does this happen before `rasa.utils.train_utils.validate_configuration_settings(self.config)`?

Reply:
It's because of deprecations we need to do it this way. For the scope of this PR, i just refactored it into a separate method and didn't change any order.
Cleaning this up is something for 3.0